python/ray/data/tests/test_dataset_formats.py

```diff
@@ -409,6 +409,117 @@ def test_read_tfrecords(ray_start_regular_shared, tmp_path):
     assert list(df["bytes_list"]) == [[b"abc", b"1234"]]
 
 
+def test_write_tfrecords(ray_start_regular_shared, tmp_path):
+    """Test that write_tfrecords writes TFRecords correctly.
+
+    Test this by writing a Dataset to a TFRecord (function under test),
+    reading it back out into a tf.train.Example,
+    and checking that the result is analogous to the original Dataset.
+    """
+
+    import tensorflow as tf
+
+    # The dataset we will write to a .tfrecords file.
+    ds = ray.data.from_items(
```

Comment:
shall we also add a negative test - throw exception when writing unsupported data type for tf.train.Feature?

Reply:
Good idea, done!